src/shapes/group.class.js

```diff
@@ -325,115 +325,43 @@
      * @return {fabric.Object} transformedObject
      */
     realizeTransform: function(object) {
-      this._moveFlippedObject(object);
-      this._setObjectPosition(object);
+      var matrix = object.calcTransformMatrix(),
+          options = fabric.util.qrDecompose(matrix),
+          center = new fabric.Point(options.translateX, options.translateY);
+      object.scaleX = options.scaleX;
```

Comment:
if you mean object.set(options) , no it does not work. Scale is often negative and is triggering an automatic flip change that creates a random flip when regrouping the objects.
 I think is part of optmization possible, but as of now it would just stop it from working.